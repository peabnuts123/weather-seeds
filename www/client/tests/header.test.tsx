import { h } from "preact";
import { render } from '@testing-library/preact'

import Header from "@app/components/header";

describe("Header component", () => {
  it("Renders consistently", () => {
    const { baseElement } = render(<Header />);
    expect(baseElement).toMatchSnapshot();
  });

  it("Header renders 3 nav items", () => {
    // Setup
    const { container } = render(<Header />);

    // Test
    const links = container.querySelectorAll('a');

    // Assert
    expect(links.length).toBe(3);
  });
});
