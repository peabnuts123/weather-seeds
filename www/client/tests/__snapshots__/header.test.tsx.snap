// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Renders consistently 1`] = `
<body>
  <div>
    <header
      class="Header"
    >
      <h1>
        Blag
      </h1>
      <nav>
        <a
          class="active"
          href="/"
        >
          Home
        </a>
        <a
          class=""
          href="/about"
        >
          About
        </a>
        <a
          class=""
          href="/limes"
        >
          Limes
        </a>
      </nav>
    </header>
  </div>
</body>
`;
